src/tests.c

```diff
@@ -400,6 +400,15 @@ void run_scratch_tests(void) {
     secp256k1_scratch_space_destroy(none, scratch);
     CHECK(ecount == 5);
 
+    /* Test that large integers do not wrap around in a bad way */
+    scratch = secp256k1_scratch_space_create(none, 1000);
+    /* try max allocation with a large number of objects */
+    CHECK(!secp256k1_scratch_max_allocation(&none->error_callback, scratch, (SIZE_MAX / (ALIGNMENT - 1)) + 1));
+    /* try allocating SIZE_MAX */
+    CHECK(ROUND_TO_ALIGN(SIZE_MAX) == 0);
```

Comment:
Couldn't `ROUND_TO_ALIGN(SIZE_MAX) == SIZE_MAX` be correct on some platforms?

Reply:
On which platforms?

Reply:
Judging from the definition of `ROUND_TO_ALIGN`, I think this happens if and only if `ALIGNMENT == 1`.

Reply:
removed line and added comment